Fixes problem with null response content in CSharp generator for non-…

jhancock93 opened the pull request:
…documented codes

The main fix is in MethodTemplate.cshtml
The remainder of the changes are auto-generated files.

This addresses issue #746 by grabbing the response content string as already done in the success case.

azuresdkci commented:
Can one of the admins verify this patch?

azurecla commented:
Hi __@jhancock93__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

stankovski commented:
@azuresdkci test this please

stankovski commented on AutoRest/Generators/CSharp/CSharp/Templates/MethodTemplate.cshtml:
```diff
@@ -228,7 +228,7 @@ else
     HttpStatusCode _statusCode = _httpResponse.StatusCode;
     cancellationToken.ThrowIfCancellationRequested();
-    string _responseContent = null;
+    string _responseContent = await _httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
```
What if the payload is a stream? You will end up with OutOfMemory exception for APIs that pipe back files or blobs.

jhancock93 replied:
Good point. This was the same code generated for the 200 case, but I guess that was for a valid model instead of a stream. I'll investigate an improved fix.

jhancock93 commented:
I've modified the original change to only read non-modeled error responses as a string in the CSharp client. So, if the default response specifies a stream model, that will work as before. I think it would be unusual for an error to return a stream, but at least this continues to handle that as before. I have also added acceptance tests to ensure that I can get the returned response content.

jhancock93 commented:
Because of the merge conflict, I can open a new pull request with code rebased on the master branch if that is preferable.

amarzavery commented:
@jhancock93 - pulling from upstream master to resolve conflicts and then pushing the changes should be fine.

jhancock93 commented:
Rebased against master and squashed commits to clean up history.

amarzavery commented:
@azuresdkci - Test this please!
